dcicutils/s3_utils.py

```diff
@@ -271,32 +269,27 @@ def health_page_get(self, health_page_key: str, *, default: Any = None, force: b
 
     ACCESS_KEYS_S3_KEY = 'access_key_admin'
 
-    def get_access_keys(self, name: str = ACCESS_KEYS_S3_KEY) -> ServerAuthDict:
-        # TODO: Make .get_access_keys() do better error checking?
-        #       I have marked the return value as ServerAuthDict because it's CLEAR that various callers
-        #       assume that values coming back from this function are in the right form. e.g.,
-        #       ff_utils.get_authentication_with_server will give a hard error if this ever returns
-        #       something with a server missing.  So presumably we mostly do the right thing here.
-        #       It would be much better to properly error-check the result here or even in .get_key().
-        #       -kmp 1-Mar-2023
+    def get_access_keys(self, name: str = ACCESS_KEYS_S3_KEY, require_key=True) -> ServerAuthDict:
+
         keys = self.get_key(keyfile_name=name)
-        if not isinstance(keys, dict):
-            raise ValueError("Remotely stored access keys are not in the expected form")
 
-        if isinstance(keys.get('default'), dict):
+        if not isinstance(keys, dict):
+            raise ValueError(f"Remotely stored {name} access keys are not in the expected form.")
+        elif 'default' in keys:
```

Comment:
I changed the conditional here. Previously it used `isinstance(keys.get('default'), dict)` but I think just `default in keys` is simpler. It's not like we want to leave `default` there when its value is `None` or anything odd like that. We still need to peel it away.